evalml/model_understanding/graphs.py

```diff
@@ -558,25 +598,23 @@ def partial_dependence(pipeline, X, features, percentiles=(0.05, 0.95), grid_res
                              "dependence is supported.")
         if not (all([isinstance(x, str) for x in features]) or all([isinstance(x, int) for x in features])):
             raise ValueError("Features provided must be a tuple entirely of integers or strings, not a mixture of both.")
+        feature_names = _get_feature_names_from_str_or_col_index(X, features)
+    else:
+        feature_names = _get_feature_names_from_str_or_col_index(X, [features])
+
     if not pipeline._is_fitted:
         raise ValueError("Pipeline to calculate partial dependence for must be fitted")
     if pipeline.model_family == ModelFamily.BASELINE:
         raise ValueError("Partial dependence plots are not supported for Baseline pipelines")
 
-    feature_list = []
-    if isinstance(features, int):
```

Comment:
Since we only check for `int` or `str` we miss the two-way case, which caused #2119